File: Robotgotchi_0001/Assets/Scripts/Web/Message.cs
using Newtonsoft.Json;

namespace DefaultNamespace
{
    public class IRequestMessage
    {
        [JsonProperty(PropertyName = "messageType")]
        public string MessageType;
        [JsonProperty(PropertyName = "payload")]
        public object Payload;
    }
    
    public class IResponseMessage
    {
        [JsonProperty(PropertyName = "messageType")]
        public string MessageType;
        [JsonProperty(PropertyName = "payload")]
        public object Payload;
    }

    public static class RequestMessageType
    {
        public const string Echo = "echo";
        public const string Login = "login";
        public const string Logout = "logout";
        public const string GlobalSettings = "globalsettings";
    }
    
    public static class ResponseMessageType
    {
        public const string UserInfo = "userinfo";
        public const string EchoResponse = "echoresponse";
        public const string GlobalSettings = "globalsettings";
    }
}

File: unity-client/src/models/message.ts
export interface IRequestMessage {
    messageType: RequestMessageType;
    payload?: any;
}

export interface IResponseMessage<T> {
    messageType: ResponseMessageType;
    payload?: T;
}

export abstract class IMessageBroker {
    abstract registerListeners(messageType: RequestMessageType, callback: () => Promise<IResponseMessage<any>>): void;
}

export abstract class IMessageListener {
    abstract listenMessage(broker: IMessageBroker): void;
}

export type RequestMessageType =
    'echo' |
    'calltestapi' |
    'getnft' |
    'getcurrentuser' |
    'logout' |
    'login';


export type ResponseMessageType =
    'userinfo' |
    'testresponse' |
    'echoresponse';

